shell/layouts/default.vue

```diff
@@ -194,224 +209,101 @@ export default {
-      });
+    currentType() {
+      return this.$route.params.resource || '';
     },
 
-    getGroups() {
-      if ( this.gettingGroups ) {
-        return;
-      }
-      this.gettingGroups = true;
-
-      if ( !this.clusterReady ) {
-        clear(this.groups);
-        this.gettingGroups = false;
-
-        return;
+    groups() {
+      if (!this.clusterReady || !this.allSchemas || !this.counts) {
+        // Referencing these computed properties
+        // to trigger the nav tree to be recomputed if they change.
+        return [];
       }
-
-      const clusterId = this.$store.getters['clusterId'];
-      const currentProduct = this.$store.getters['productId'];
-      const currentType = this.$route.params.resource || '';
       let namespaces = null;
 
+      // Always show cluster-level types, regardless of the namespace filter
       if ( !this.$store.getters['isAllNamespaces'] ) {
         const namespacesObject = this.$store.getters['namespaces']();
 
         namespaces = Object.keys(namespacesObject);
       }
-
-      // Always show cluster-level types, regardless of the namespace filter
-      const namespaceMode = 'both';
       const out = [];
-      const loadProducts = this.isExplorer ? [EXPLORER] : [];
+
       const productMap = this.activeProducts.reduce((acc, p) => {
         return { ...acc, [p.name]: p };
       }, {});
 
-      if ( this.isExplorer ) {
-        for ( const product of this.activeProducts ) {
-          if ( product.inStore === 'cluster' ) {
-            addObject(loadProducts, product.name);
-          }
-        }
-      }
-
-      // This should already have come into the list from above, but in case it hasn't...
-      addObject(loadProducts, currentProduct);
-
-      for ( const productId of loadProducts ) {
+      for (const productId of this.loadedProducts) {
         const modes = [BASIC];
 
-        if ( productId === NAVLINKS ) {
+        if (productId === NAVLINKS) {
           // Navlinks produce their own top-level nav items so don't need to show it as a product.
           continue;
         }
 
-        if ( productId === EXPLORER ) {
+        if (productId === EXPLORER) {
           modes.push(FAVORITE);
           modes.push(USED);
         }
 
-        for ( const mode of modes ) {
-          const types = this.$store.getters['type-map/allTypes'](productId, mode) || {};
-
-          const more = this.$store.getters['type-map/getTree'](productId, mode, types, clusterId, namespaceMode, namespaces, currentType);
-
-          if ( productId === EXPLORER || !this.isExplorer ) {
+        for (const mode of modes) {
+          const types =
+            this.$store.getters['type-map/allTypes'](productId, mode) || {};
+
+          // Because the "clusterId", "namespaces" and "namespaceMode"
+          // computed properties are dependencies of "groups", the nav tree will be
+          // recomputed if they change.
+          const more = this.$store.getters['type-map/getTree'](
```

Comment:
This line references a bunch of computed properties that will now trigger updates.